Engine/platform/windows/gfx/ali3dd3d.cpp

```diff
@@ -1339,6 +1340,7 @@ void D3DGraphicsDriver::_renderAndPresent(GlobalFlipType flip, bool clearDrawLis
 {
   _render(flip, clearDrawListAfterwards);
   direct3ddevice->Present(NULL, NULL, NULL, NULL);
+  DelayUntilEndOfFrame();
```

Comment:
I honestly don't like this change, I don't think it's a proper task for this class to do delays.

Reply:
IMO it was already done wrong that there are fade loops inside the renderer classes, perhaps it would be better to "unwrap" these somehow instead.

Reply:
I thought putting it there might make sense since there's already a delay for vsync. I can move the `DelayUntilEndofFrame`s outside the render function, there might just be a few more of them :)

I agree with the fade effects needing to be moved out of the graphics driver but I that's possibly for a different PR. It would be tricky to ensure it's correct across the different gfx drivers.